crates/viewer/re_renderer/src/video/decoder/native_decoder.rs

```diff
@@ -18,22 +18,25 @@ struct DecoderOutput {
     error: Option<TimedDecodingError>,
 }
 
-/// Native AV1 decoder
-pub struct Av1VideoDecoder {
-    decoder: re_video::av1::Decoder,
+/// Native video decoder
+pub struct NativeDecoder {
+    decoder: re_video::decode::AsyncDecoder,
     decoder_output: Arc<Mutex<DecoderOutput>>,
 }
 
-impl Av1VideoDecoder {
-    pub fn new(debug_name: String) -> Result<Self, DecodingError> {
+impl NativeDecoder {
+    pub fn new(
+        debug_name: String,
+        sync_decoder: Box<dyn re_video::decode::SyncDecoder + Send>,
```

Comment:
found it odd at first that all `NativeDecoder`s host a `SyncDecoder` internally. But from the looks of it this ~gonna hold up at least for a a few weeks 😁~ this is what we're dealing with :)